features/mbedtls/targets/TARGET_STM/sha256_alt.c

```diff
@@ -0,0 +1,141 @@
+/*
+ *  sha256_alt.c for SHA256 HASH
+ *******************************************************************************
+ * Copyright (c) 2017, STMicroelectronics
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+#include "mbedtls/sha256.h"
```

Comment:
On second thought and after discussing it with @yanesca, this additional guard should NOT be included in the code right now. However, I was trying to think ahead as it will be required when [this mbed TLS issue](https://github.com/ARMmbed/mbedtls/issues/660) is fixed.